### ISSUE TITLE
Backport Record support to the 1.x

### ISSUE THREAD

**Issue by Verdent** [eclipse-ee4j/yasson#528]
Title says it all :-)

**Comment by lostiniceland** [comment]
@Verdent I started with this today, but after cloning I realised that both `master` and `1.x` have failing tests. Do I have to pass some arguments to `mvn package` wo make it work?
```

[ERROR] Failures: 
[ERROR]   DocumentationExampleTest.testDateNumberFormats1:368 expected: <{"birthDate":"07.08.1999","name":"Jason Bourne","salary":"123.46"}> but was: <{"birthDate":"07.08.1999","name":"Jason Bourne","salary":"123,46"}>
[ERROR]   DocumentationExampleTest.testDateNumberFormats2:398 expected: <123.45678> but was: <123>
[ERROR] Errors: 
[ERROR]   CdiInjectionTest.testInjectionAndCleanup » Weld WELD-001524: Unable to load pr...
[INFO] 
[ERROR] Tests run: 463, Failures: 2, Errors: 1, Skipped: 1

```

**Comment by Verdent** [comment]
Hi @lostiniceland , thank you very much for volunteering to do that. Unfortunately the record handling needed some fixing and I am having PR opened for that and hopefully it will be merged and backported this week. 

See #530

Now in terms of the failing tests. This happens usually when your local setup is running on different localization than EN. Is that also your issue? I need to fix those test to be more universal, or to set everything up properly, but I didn't have time for it yet :-)

Sure only passing PRs are merged and none other would be allowed. 

What confuses me a bit though is that failing CdiInjectionTest. I have never encountered that before. May I ask you to do a little diggings around it? :-)

**Comment by lostiniceland** [comment]
Yes, I am running German localisation, which seems to be the culprit here. I will try to do some digging and maybe I can fix those tests.

Regarding the backport: I just cherry-picked the one from 2.x and had to adjust some paths and static-methods. The record-tests seemed to work (but I didnt fully checked because I wanted the baseline-tests to run first)

**Comment by Verdent** [comment]
Records handling is supported by the first version, but it is actually having problems in dealing with edited records such as multiple constructors, JsonbCreator included, extra methods added etc. :-) That had to be fixed and that's what my mentioned PR is fixing.

**Comment by Verdent** [comment]
@lostiniceland I have fixed (or better to say... made it working:-D ) that CDI test. It was failing due to higher version of java. Complete backport with the cdi test fix is here #531

**Comment by Verdent** [comment]
@lostiniceland Just letting you know that Yasson 1.0.11 and 2.0.4 are being released today to the maven central. :-)
